### PR TITLE
Fix linux compile and some double-discards

### PR DESCRIPTION
**Pull request by JJL772**
Fixes the linux compile and some double discards that were causing crashes on exit, at least on Nvidia's linux driver. 

Probably needs testing on windows because I make some changes to the parameters passed to MSVC.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,35 @@
...
 
 # Definitions
 if (CMAKE_BUILD_TYPE MATCHES Debug)
     add_definitions(-DDEBUG)
+    if(UNIX)
+        # On UNIX, build with debug friendly optimizations and debug symbols
+        # On windows, just disable optimizations entirely
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Og -g")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Og -g")
+    else()
```
Comment (craftablescience) [inline CMakeLists.txt:13]: `elseif(MSVC)`

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,12 +1,35 @@
...
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /Od")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Od")
+    endif()
 elseif(CMAKE_BUILD_TYPE MATCHES Release)
     add_definitions(-DRELEASE)
+    # Build with max optimizations and don't omit stack ptr for debugging reasons
+    if(UNIX)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -fno-omit-frame-pointer")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -fno-omit-frame-pointer")
+    else()
```
Comment (craftablescience) [inline CMakeLists.txt:23]: `elseif(MSVC)`

**Review by craftablescience — CHANGES_REQUESTED** [review]
This is awesome, thanks for the Linux support!
Just noticed two things on the Windows end, adding these flags breaks MinGW. Works fine for MSVC though.

File: CMakeLists.txt (hunk 1 above)
Comment (craftablescience) [reply CMakeLists.txt:13]: Of course GitHub cuts off the part I'm referring to

**Comment by craftablescience** [comment]
I'll just make these changes after merging

**Review by craftablescience — APPROVED** [review]

**Comment by JJL772** [comment]
Oh whoops, thanks for making the changes lol

**Pull request merged**